samza-azure/src/main/java/org/apache/samza/coordinator/AzureJobCoordinator.java

```diff
@@ -367,7 +367,8 @@ private void doOnProcessorChange(List<String> currentProcessorIds) {
     // Generate the new JobModel
     GrouperMetadata grouperMetadata = new GrouperMetadataImpl(Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap());
     JobModel newJobModel =
-        JobModelManager.readJobModel(this.config, Collections.emptyMap(), streamMetadataCache, grouperMetadata);
+        JobModelCalculator.INSTANCE.calculateJobModel(this.config, Collections.emptyMap(), streamMetadataCache,
+            grouperMetadata);
```

Comment:
It is easier to do dependency injection this way for testing purposes.
For this class, the call is just treated like a static utility method call, but other classes can inject a `JobModelCalculator` so that it can be mocked out in testing.